signature-aggregator/aggregator/aggregator.go

```diff
@@ -0,0 +1,507 @@
+	m := response.Message()
+	rcvReqID, ok := message.GetRequestID(m)
+	if !ok {
+		// This should never occur, since inbound message validity is already checked by the inbound handler
+		s.logger.Error("Could not get requestID from message")
+		return nil, false, nil
+	}
+	nodeID := response.NodeID()
+	if !sentTo.Contains(nodeID) || rcvReqID != requestID {
+		s.logger.Debug("Skipping irrelevant app response")
+		return nil, false, nil
+	}
+
+	// If we receive an AppRequestFailed, then the request timed out.
+	// This is still a relevant response, since we are no longer expecting a response from that node.
+	if response.Op() == message.AppErrorOp {
+		s.logger.Debug("Request timed out")
+		return nil, true, nil
+	}
+
+	validator, vdrIndex := connectedValidators.GetValidator(nodeID)
+	signature, valid := s.isValidSignatureResponse(unsignedMessage, response, validator.PublicKey)
+	if valid {
+		s.logger.Debug(
+			"Got valid signature response",
+			zap.String("nodeID", nodeID.String()),
+			zap.Uint64("stakeWeight", validator.Weight),
+			zap.String("warpMessageID", unsignedMessage.ID().String()),
+			zap.String("sourceBlockchainID", unsignedMessage.SourceChainID.String()),
+		)
+		signatureMap[vdrIndex] = signature
+		accumulatedSignatureWeight.Add(accumulatedSignatureWeight, new(big.Int).SetUint64(validator.Weight))
+	} else {
+		s.logger.Debug(
+			"Got invalid signature response",
+			zap.String("nodeID", nodeID.String()),
+			zap.Uint64("stakeWeight", validator.Weight),
+			zap.String("warpMessageID", unsignedMessage.ID().String()),
+			zap.String("sourceBlockchainID", unsignedMessage.SourceChainID.String()),
+		)
+		return nil, true, nil
+	}
+
+	// As soon as the signatures exceed the stake weight threshold we try to aggregate and send the transaction.
+	if utils.CheckStakeWeightPercentageExceedsThreshold(
+		accumulatedSignatureWeight,
+		connectedValidators.TotalValidatorWeight,
+		quorumPercentage,
+	) {
+		aggSig, vdrBitSet, err := s.aggregateSignatures(signatureMap)
+		if err != nil {
+			msg := "Failed to aggregate signature."
+			s.logger.Error(
+				msg,
+				zap.String("sourceBlockchainID", unsignedMessage.SourceChainID.String()),
+				zap.String("warpMessageID", unsignedMessage.ID().String()),
+				zap.Error(err),
+			)
+			return nil, true, fmt.Errorf("%s: %w", msg, err)
+		}
+
+		signedMsg, err := avalancheWarp.NewMessage(
+			unsignedMessage,
+			&avalancheWarp.BitSetSignature{
+				Signers:   vdrBitSet.Bytes(),
+				Signature: *(*[bls.SignatureLen]byte)(bls.SignatureToBytes(aggSig)),
+			},
+		)
+		if err != nil {
+			msg := "Failed to create new signed message"
+			s.logger.Error(
+				msg,
+				zap.String("sourceBlockchainID", unsignedMessage.SourceChainID.String()),
+				zap.String("warpMessageID", unsignedMessage.ID().String()),
+				zap.Error(err),
+			)
+			return nil, true, fmt.Errorf("%s: %w", msg, err)
+		}
+		return signedMsg, true, nil
+	}
+	// Not enough signatures, continue processing messages
+	return nil, true, nil
```

Comment:
We can combine these by declaring `signedMsg = nil` above this `if` block.

Reply:
For me it's more readable to keep the control flow as is with the comment. Makes it clear that the happy case terminates inside the if statement loop and that if you are outside you didn't get enough signatures. It's also not reducing LOC since it would be trading adding `signedMsg=nil` above for the return value. 

If you strongly prefer it I will make the change though

Reply:
Yeah, I think I prefer explicitly returning nil